pallets/api/src/messaging/mod.rs

```diff
@@ -316,40 +326,60 @@ pub mod pallet {
 			// in pallet-xcm. As a result, we must handle timeouts in the pallet.
 			// Iterate through the queries that have expired and update their status.
 			let mut weight: Weight = Zero::zero();
+			let mut query_ids = sp_std::vec![];
 			for (origin, message_id) in XcmQueryTimeouts::<T>::get(n) {
 				weight = weight.saturating_add(DbWeightOf::<T>::get().reads_writes(2, 1));
 				Messages::<T>::mutate(origin, message_id, |maybe_message| {
 					if let Some(Message::XcmQuery { query_id, deposit, .. }) =
 						maybe_message.as_mut()
 					{
+						query_ids.push(*query_id);
 						*maybe_message =
 							Some(Message::XcmTimeout { query_id: *query_id, deposit: *deposit });
 					}
 				})
 			}
 
+			if !query_ids.is_empty() {
```

Comment:
just asking a sanity question (answer is likely yes). In this `on_initialize` hook, is the deposit charged large enough to dissuade someone from spamming XCM requests thus causing a large amount of "free" mutations in this hook?

Reply:
Since they pay upfront for the response, that will cover easily the cost of this